overlord/snapstate/backend/snapdata.go

```diff
@@ -108,9 +123,37 @@ func copySnapData(oldSnap, newSnap *snap.Info) (err error) {
 	return nil
 }
 
+func trashPath(path string) string {
+	return path + ".old"
+}
+
+// trash moves path aside
+func trash(path string) error {
+	return os.Rename(path, trashPath(path))
```

Comment:
Do we need to deal with the case of the trash already existing here? Otherwise, what happens if we crash with the trash in place.. would we never recover?